### DISCUSSION TITLE
Conversion on windows produces bitmaps that won't open in Crystal Reports.

### DISCUSSION THREAD

**Discussion by IainSharp** [ImageMagick/ImageMagick#5985] [Help] [answered]
### ImageMagick version

7.1.0-57

### Operating system, version and so on

Windows Server 2019

### Description

I have a PNG with transparency from a web source, when I use magick to convert this to a bmp the file shows as a preview in explorer briefly, but then switches to entirely black. It won't import into Crystal Reports (Not a valid image file). 
If I load it into paint, and then save without making any changes, the file will preview and import into Crystal correctly. 

I believe this is a file format/encoding issue but do not know what (if any) options I can apply to the conversion to set up a usable file format. 

I am currently using simply
magick file.png fileout.bmp

If I take the resaved (from paint) and therefore useable (to me) bitmap file, and crop it using 
magick fileout.bmp -crop x368 fileout2.bmp
it produces a file which will preview in explorer, but not import into crystal unless I open it in Paint and resave it. 

Any thoughts? 
Iain

**Comment by dlemstra** [comment]
Might be a good idea to share your input image so we can see what is happening? And can you reproduce the issue when you crop it directly?

**Reply by IainSharp** [reply]
[iii-0.zip](https://github.com/ImageMagick/ImageMagick/files/10456295/iii-0.zip)

Attached is a zip containing the original png, the iii-0.bmp cropped directly from it, and the iii-0sv.bmp which has been loaded in paint and resaved. 
I found the -alpha remove command in the mean time which has removed the transparency (I didn't think bitmaps could have transparency?)
Thanks

**Reply by IainSharp** [reply]
Sorry, to be clear, the file is still not importable into Crystal, even with the transparency removed. 
I can import the original into crystal as a recognised file type, it's just entirely black as the transparency is replaced with black.

**Comment by IainSharp** [comment]
Okay, I FAFO and have come up with the command line. 
magick mxsignPN-480177-1.PNG -crop x368 -alpha remove -resize 12%  iii.png
Which creates a png I can then use in Crystal. 
I would still be interested to know why the bitmap output is unusable, as I can't spot any difference between the two bmp files.

**Reply by fmw42** [reply]
I think Imagemagick produces BMP4 format. Perhaps your good format for Crystal is BMP3 or BMP2.

**Comment by IainSharp** [comment]
I have tried
-format BMP2
-format BMP3
Without success. And the output files are suspiciously small for bitmaps (but possible I suppose).
Is this significantly different from bmp2:output.bmp?
Regards,
Iain

**Comment by snibgo** [comment, marked as answer]
> magick file.png fileout.bmp
I have tried
-format BMP2
-format BMP3
Without success.

But that isn't how to set the output image format. Try:
```
magick file.png BMP2:fileout.bmp
magick file.png BMP3:fileout.bmp
```

**Reply by IainSharp** [reply]
So it would appear that **magick file.PNG -crop x368 -alpha remove -resize 12% bmp3:file.bmp** is the necessary output. 
It appears -format only works with **mogrify**. 
Thanks

**Comment by snibgo** [comment]
"`-format`" works on both "`magick mogrify`" and plain "`magick`", but does a different job on each. See https://imagemagick.org/script/command-line-options.php#format
